control/freqplot.py

```diff
@@ -117,65 +117,100 @@ def bode_plot(syslist, omega=None, dB=None, Hz=None, deg=None,
     # If argument was a singleton, turn it into a list
     if (not getattr(syslist, '__iter__', False)):
         syslist = (syslist,)
-
-    mags, phases, omegas = [], [], []
+    
+    if omega is None:
+        # Select a default range if none is provided
+        omega = default_frequency_range(syslist, Hz=Hz)
+    elif (isinstance(omega, tuple) or isinstance(omega, list)) and len(omega) == 2:
```

Comment:
do you want to also support numpy.ndarray for omega here?  isinstance takes a tuple as second arg, so this could be `isinstance(omega, (tuple,list,np.ndarray))`.  There doesn't seem to be a good general way to check for a sequence (see, e.g., http://stackoverflow.com/a/2937122/1008142 )